### PR TITLE
Fix bug where progress indicator doesn't always complete

### PR DESCRIPTION
**Pull request by ozabluda**
Before:
16256/17820 [==========================>...] - ETA: 0s
After:
17820/17820 [==============================] - ETA: 0s

Also improve code readability if => elif. No semantics change.

### CODE REVIEW

**Comment by ozabluda** [comment]
Just noticed that a similar fix was done in https://github.com/fchollet/keras/pull/6536/files then reverted in
https://github.com/fchollet/keras/commit/1a89b13cb4ac52b7dad40e671b618e42921a4c48 with comment:
> this appears to be causing a 2x slowdown of the test suite with Python 2.7. Unclear why. I reverted it.
I checked on GitHub Travis, and it the slowdown doesn't seem to be happening anymore.

**Comment by ozabluda** [comment]
I will change progbar interval from 0.05 to 0.1 to see if it makes a difference in test running times in Travis. 50 ms is too fast for a human anyway.

**Comment by ozabluda** [comment]
Unfortunately (as expected), one can't really tell the difference this way. Still, I think 100ms is a better interval than 50 ms.

File: keras/utils/generic_utils.py [hunk 1]
```diff
@@ -292,7 +292,7 @@ def update(self, current, values=None, force=False):
 
         now = time.time()
         if self.verbose == 1:
-            if not force and (now - self.last_update) < self.interval:
+            if not force and (now - self.last_update) < self.interval and current < self.target:
```
[the remark this answers is not in the document]
Comment (ozabluda) [reply keras/utils/generic_utils.py]: The reason I didn't make it shorter already is that I saw that you don't want \ and I didn't find (sorry) an example in the code of an acceptable style, and didn't find guidance in pep8. I can break it after 'and', but it's kinda weird.

[the remark this answers is not in the document]
Comment (fchollet) [reply keras/utils/generic_utils.py]: You can use parentheses to structure code into several lines.

**Pull request merged**